### ISSUE TITLE
Villagers buying books not working properly

### ISSUE THREAD

**Issue by Laxliet** [GeyserMC/Geyser#1599]
**Describe the bug**
Simple. Villagers aren't giving enchanted books from trade.

**To Reproduce**
Steps:
1-Find a librarian villager who sells books
2-Try to trade a enchanted book
3-You will see the issue

**Expected behavior**
Villagers need to give enchanted books

**Screenshots / Videos**
I don't think you need screenshots or videos for that. Just take your one minute and test it.

**Server Version**
git-Paper-294

**Geyser Version**
Build 496

**Minecraft: Bedrock Edition Version**
1.16.100

**Additioal Context**
https://dump.geysermc.org/a4MwoEgTBY7ljt030zriRLVZFoWC4Yhv

not because of my plugins. I tested it on a clean server too.

**Comment by DoctorMacc** [comment]
Looks like we send an inventory packet that the server doesn't like. I'll look into this more tomorrow.

**Comment by Laxliet** [comment]
Okay. Btw if u try too much books are glitching too, they are filling inventory every slot and being unstackable. I'll record a video for that

**Comment by DoctorMacc** [comment]
Looks like the issue here is a little deceptive. Making a custom villager with that buys emeralds for an enchanted book works fine, which one can verify by trading with this villager:

```
/summon villager ~ ~-2 ~ {NoAI:0b,Willing:1b,LastRestock:0,Xp:0,VillagerData:{level:1,profession:"minecraft:librarian",type:"minecraft:desert"},Offers:{Recipes:[{rewardExp:1b,maxUses:2147483647,uses:0,xp:1,priceMultiplier:1f,specialPrice:1,demand:1,buy:{id:'minecraft:emerald',Count:10b},sell:{id:'minecraft:enchanted_book',Count:1b,tag:{StoredEnchantments:[{id:'minecraft:loyalty',lvl:1s}]}}}]},Brain:{memories:{"minecraft:meeting_point":{value:{pos:[I;0,0,0],dimension:"minecraft:overworld"}},"minecraft:home":{value:{pos:[I;0,0,0],dimension:"minecraft:overworld"}},"minecraft:job_site":{value:{pos:[I;0,0,0],dimension:"minecraft:overworld"}}}}}```

Rather, it's the book being an input causing the issue.

**Comment by Laxliet** [comment]
Yea, I can use commands. but how about players? I'm hosting a survival server. And they will make their own villagers ???

**Comment by DoctorMacc** [comment]
I'm noting down what I'm finding about this issue. Ideally I'll find the root cause of this issue instead.

**Comment by Laxliet** [comment]
Oh nice. I hope this will fix in soon ._. Good luck

**Comment by DoctorMacc** [comment]
@Laxliet can you please try #1605? You may need to get new books, depending on your situation. Let us know if you need help downloading a PR.

**Comment by Laxliet** [comment]
Sure, let me check

**Comment by Laxliet** [comment]
Issue fixed with #1605
